### PR TITLE
refector wait.HTTPStrategy

### PR DESCRIPTION
**Pull request by islishude**
- add method and body for request
- add tls.Config for https
- use time.After instead of time.Sleep

### CODE REVIEW

**Comment by mdelapenya** [comment]
Hey @islishude, thank you for this nice contribution! Do you think it would be possible to add a test for the new behavior, specially the TLSConfig part?

Thanks!

**Comment by islishude** [comment]
OK,I will add test cases for it

**Comment by islishude** [comment]
Hi,the tests pass,please review @mdelapenya

File: wait/http_test.go [hunk 1]
```diff
@@ -25,7 +36,88 @@ func ExampleHTTPStrategy() {
...
+			ExpectContinueTimeout: 1 * time.Second,
+		},
+	}
+	resp, err := client.Get(fmt.Sprintf("https://%s:%s", host, port.Port()))
+	if err != nil {
+		t.Error(err)
+		return
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode != http.StatusOK {
```
Comment (mdelapenya) [inline wait/http_test.go:119]: What do you think about adding an assertion for the `pong` response? We'd like to check that the response matcher works as intended.

Comment (islishude) [reply wait/http_test.go:119]: this variable indicates a state code, and I consider the value of `pong` to be a success state.

in practice, the server can return a response body with a state code representing the current server state,so we can know if the server is ready based on what is specifically returned.

Comment (mdelapenya) [reply wait/http_test.go:119]: :) I meant about adding another assertion for the expected response (pong), which is part of the waitFor condition predefined when setting up the test in the response matcher. Of course this is not a requirement by the way, so I'm not opposing to it.

In the same hand, the PR as it's now could be merged, so I want to thank you for your contribution

**Review by mdelapenya — APPROVED** [review]

**Pull request merged**